crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -119,15 +119,14 @@ impl BaseClient {
     ///   sync.
     /// * `previous_events_provider` - Timeline events prior to the current
     ///   sync.
-    /// * `from_simplified_sliding_sync` - Whether the `response` comes from
-    ///   simplified sliding sync (Simplified MSC3575), or sliding sync
-    ///   (MSC3575).
+    /// * `from_msc4186` - Whether the `response` comes from simplified sliding
+    ///   sync (MSC4186) or sliding sync (MSC3575).
     #[instrument(skip_all, level = "trace")]
     pub async fn process_sliding_sync<PEP: PreviousEventsProvider>(
         &self,
         response: &http::Response,
         previous_events_provider: &PEP,
-        from_simplified_sliding_sync: bool,
+        from_msc4186: bool,
```

Comment:
uber-nit: `with_msc4186`? The msc itself isn't a source of sync data, `from_..._sync` made sense up to now, but not the case with the msc number.

Reply:
I agree it creates confusion. Not a big fan of `with_…` but I can't come up with a better naming. Thanks for the proposal!